src/query/engine.ts

```diff
@@ -344,6 +362,76 @@ export function executeTask(query: Query, origin: string, index: FullIndex, sett
     });
 }
 
+/** Shared execution code which just takes in arbitrary data, runs operations over it, and returns it + per-row errors. */
+export function filterTasks(page: PageMetadata, pages: Pagerow[], rows: Task[], context: Context, ops: QueryOperation[]): Result<TaskExecution, string> {
```

Comment:
Instead of re-implementing all of the operators, it might be easier to just cast each task into a Pagerow and then use the existing `executeCore` over them; you can pre-process each task and add the default fields if they don't exist, and then pull out the task objects afterwards and cast them back to tasks.

Reply:
Great call! I'd been noodling about how to reuse the existing function, and was thinking about a shared interface or inheritance. Is it as simple as just `as Pagerow; ...; as Task`? I suppose the constructors would also work with their `Partial` implementation.